app/lib/cashu/animated-qr-code/use-animated-qr-encoder.tsx

```diff
@@ -0,0 +1,54 @@
+import type { UREncoder } from '@gandlaf21/bc-ur';
+import { useState } from 'react';
+import { useInterval } from 'usehooks-ts';
+
+type Props = {
+  text: string;
+  intervalMs?: number;
+};
+
+const maxFragmentLength = 200;
+const firstSequenceNumber = 0;
+
+function bufferFromUtf8(text: string) {
+  return new Uint8Array(text.split('').map((c) => c.charCodeAt(0)));
+}
+
+export function useAnimatedQREncoder({ text, intervalMs = 200 }: Props) {
+  const [fragment, setFragment] = useState<string>('');
+  const [encoder, setEncoder] = useState<UREncoder | null>(() => {
+    if (typeof window !== 'undefined') {
+      // bc-ur has cborg as a dependency which fails to import on the server
+      import('@gandlaf21/bc-ur').then(({ UR, UREncoder }) => {
+        const messageBuffer = bufferFromUtf8(text);
+        // @ts-expect-error - expects Buffer, but Uint8Array works
+        const ur = UR.fromBuffer(messageBuffer);
```

Comment:
another option would be to install [buffer](https://www.npmjs.com/package/buffer) if we don't want to ignore the type error